streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -119,15 +121,43 @@ public final void removeAllTaskLevelSensors(final String taskName) {
         }
     }
 
+    public Sensor nodeLevelSensor(final String taskName, final String processorNodeName, final String sensorName, final Sensor.RecordingLevel recordingLevel, final Sensor... parents) {
+        final String key = threadName + "." + taskName + ".node." + processorNodeName;
+        synchronized (nodeLevelSensors) {
+            if (!nodeLevelSensors.containsKey(key)) {
+                nodeLevelSensors.put(key, new LinkedList<>());
+            }
+
+            final String fullSensorName = key + "." + sensorName;
+
+            final Sensor sensor = metrics.sensor(fullSensorName, recordingLevel, parents);
+
+            nodeLevelSensors.get(key).push(fullSensorName);
+
+            return sensor;
+        }
+    }
+
+    public final void removeAllNodeLevelSensors(final String taskName, final String processorNodeName) {
+        final String key = threadName + "." + taskName + ".node." + processorNodeName;
+        synchronized (nodeLevelSensors) {
+            if (nodeLevelSensors.containsKey(key)) {
+                while (!nodeLevelSensors.get(key).isEmpty()) {
+                    metrics.removeSensor(nodeLevelSensors.get(key).pop());
+                }
+            }
+        }
+    }
+
     public final Sensor cacheLevelSensor(final String taskName,
                                          final String cacheName,
                                          final String sensorName,
                                          final Sensor.RecordingLevel recordingLevel,
                                          final Sensor... parents) {
-        final String key = threadName + "." + taskName + "." + cacheName;
+        final String key = threadName + "." + taskName + ".cache." + cacheName;
```

Comment:
now that there are other sensors with the same path length, it seems safer to namespace them all.

Changing the sensor name shouldn't impact anything.

Reply:
this is the name pattern as for `.node.` -- should we have  method that computes the name passing in `node` or `cache` ?

Reply:
¯\_(ツ)_/¯ 

It's really trivial logic, so I'm not sure of the value of adding an extra layer of indirection.

Reply:
It's not about being trivial or not, but about consistency. Code duplication has the danger to only update one copy of the code...

Reply:
I opted for direct string concatenation here, since it's flatter, and therefore easier to reason about, and in this case, I thought that a separate method would basically offer the same pros and cons as concatenation.

That's also why I flattened the other sensor names, the method calls (two of them) were obscuring the fact that one of the portions was totally unused. Once it's flat, you can see directly what string is produced.

Also contributing factors: we need to append one more part to the key after the fact to get the "fullSensorName", and each of these are completely internal and need to be declared in exactly two places, the xLevelSensor and the removeAllXLevelSensors methods.

It seems like concatenation is slightly better, but I'm open to changing it if you feel strongly about it.

Reply:
I've reconsidered this stance, and added methods to create the "prefix" key for each level of sensor.